src/openloco/localisation/stringmgr.cpp

```diff
@@ -1,23 +1,563 @@
+
+        return str;
+    }
+
+    // TODO: decltype(value)
+    static char* format_comma(uint32_t value, char* buffer)
+    {
+        registers regs;
+        regs.eax = value;
+        regs.edi = (uint32_t) buffer;
+
+        call(0x00495F35, regs);
+        return (char*) regs.edi;
+    }
+
+    // TODO: decltype(value)
+    static char* format_int(uint32_t value, char* buffer)
+    {
+        registers regs;
+        regs.eax = value;
+        regs.edi = (uint32_t) buffer;
+
+        call(0x495E2A, regs);
+        return (char*) regs.edi;
+    }
+
+    // TODO: decltype(value)
+    static char* formatNumeric_4(uint16_t value, char* buffer)
+    {
+        registers regs;
+        regs.eax = (uint32_t) value;
+        regs.edi = (uint32_t) buffer;
+
+        call(0x4963FC, regs);
+        return (char*) regs.edi;
+    }
+
+    // TODO: decltype(value)
+    static char* format_comma2dp32(uint32_t value, char* buffer)
+    {
+        registers regs;
+        regs.eax = value;
+        regs.edi = (uint32_t) buffer;
+
+        call(0x4962F1, regs);
+        return (char*) regs.edi;
+    }
+
+    static char* formatDayMonthYearFull(uint32_t value, char* buffer)
+    {
+        registers regs;
+        regs.eax = (uint32_t) value;
+        regs.edi = (uint32_t) buffer;
+
+        call(0x495CB5, regs);
+        return (char*) regs.edi;
+    }
+
+    static char* formatMonthYearFull(uint32_t value, char* buffer)
+    {
+        registers regs;
+        regs.eax = (uint32_t) value;
+        regs.edi = (uint32_t) buffer;
+
+        call(0x495D77, regs);
+        return (char*) regs.edi;
     }
 
-    void format_string(char* buffer, string_id id, void* args)
+    static char* formatMonthYearAbbrev_0(uint32_t value, char* buffer)
     {
         registers regs;
-        regs.eax = id;
-        regs.edi = (uint32_t)buffer;
-        regs.ecx = (uint32_t)args;
-        call(0x004958C6, regs);
+        regs.eax = (uint32_t) value;
+        regs.edi = (uint32_t) buffer;
+
+        call(0x495DC7, regs);
+        return (char*) regs.edi;
+    }
+
+    static char* format_string(char* buffer, string_id id, argswrapper &args);
+
+    static char* format_string_part(char* buffer ,const char* sourceStr, argswrapper &args)
+    {
+        while (uint8_t ch = *sourceStr++)
+        {
+            if (ch <= 0x1F)
+            {
+                if (ch == 0)
+                {
+                    *buffer = '\0';
+                    return buffer;
+                }
+                else if (ch <= 4)
+                {
+                    *buffer = ch;
+                    buffer++;
+
+                    ch = *sourceStr;
+                    sourceStr++;
+
+                    *buffer = ch;
+                    buffer++;
+                }
+                else if (ch <= 16)
+                {
+                    *buffer = ch;
+                    buffer++;
+                }
+                else
+                {
+                    if (ch <= 22)
```

Comment:
`if (ch > 22)`

Also, just make this another case of the parent if, instead of a subcase.